core/src/idx/trees/hnsw/mod.rs

```diff
@@ -242,15 +249,22 @@ where
 
 	fn search_ep(&self, pt: &SharedVector) -> Option<(f64, ElementId)> {
 		if let Some(mut ep_id) = self.enter_point {
-			let mut ep_dist =
-				self.elements.get_distance(pt, &ep_id).unwrap_or_else(|| unreachable!());
-			for layer in self.layers.iter().rev() {
-				(ep_dist, ep_id) = layer
-					.search_single(&self.elements, pt, ep_dist, ep_id, 1)
-					.peek_first()
-					.unwrap_or_else(|| unreachable!());
+			if let Some(mut ep_dist) = self.elements.get_distance(pt, &ep_id) {
+				for layer in self.layers.iter().rev() {
+					if let Some(ep_dist_id) =
+						layer.search_single(&self.elements, pt, ep_dist, ep_id, 1).peek_first()
+					{
+						(ep_dist, ep_id) = ep_dist_id;
+					} else {
+						#[cfg(debug_assertions)]
+						unreachable!()
+					}
+				}
+				Some((ep_dist, ep_id))
+			} else {
+				#[cfg(debug_assertions)]
+				unreachable!()
```

Comment:
This part has broken the release build, because else branch does not return `None`